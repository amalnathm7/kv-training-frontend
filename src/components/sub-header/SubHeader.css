.sub-header {
    display: flex;
    align-items: center;
    justify-content: space-between;
    background-color: white;
    box-shadow: rgba(0, 0, 0, 0.1) 0px 4px 6px -1px, rgba(0, 0, 0, 0.06) 0px 2px 4px -1px;
    position: absolute;
    top: 110px;
    left: 320px;
    right: 20px;
    height: 80px;
    padding-left: 20px;
    padding-right: 20px;
    border-radius: 10px;
}

.sub-header-label {
    font-size: 24px;
    font-weight: bold;
}

.action-button {
    display: flex;
    align-items: center;
    height: 50px;
    background-color: #EAF9FF;
    border-radius: 30px;
    cursor: pointer;
}

.sub-header-action-icon-container {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 50px;
    height: 50px;
    border-radius: 100%;
    background-color: #03AEEE;
    margin-right: 20px;
}

.sub-header-action-label {
    font-size: 16px;
    margin-right: 30px;
    cursor: pointer;
}

.sub-header-action-icon {
    height: 20px;
    width: 20px;
}
