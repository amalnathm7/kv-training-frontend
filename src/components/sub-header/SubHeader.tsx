import React from 'react';
import './SubHeader.css';

export type SubHeaderPropsType = {
  label: string;
  primaryActionLabel: string;
  primaryActionIcon: string;
  primaryAction: () => void,
    secondaryActionLabel?: string,
    secondaryActionIcon?: string,
    secondaryAction?: () => void;
  searchClicked?: boolean;
};

const SubHeader: React.FC<SubHeaderPropsType> = (props) => {
    return <div className="sub-header">
        <label className="sub-header-label">{props.label}</label>
        {props.primaryActionLabel.length > 0 && <div className="action-button" onClick={props.primaryAction}>
            <div className="sub-header-action-icon-container">
                <img className="sub-header-action-icon" src={"/assets/icons/" + props.primaryActionIcon}></img>
            </div>
            <label className="sub-header-action-label">{props.primaryActionLabel}</label>
        </div>}
    </div>;
};

export default SubHeader;
