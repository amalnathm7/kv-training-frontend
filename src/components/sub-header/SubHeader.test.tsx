import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import SubHeader, { SubHeaderPropsType } from './SubHeader';

describe('SubHeader Props Test', () => {
  test('If SubHeader rendered correctly', () => {
    const props: SubHeaderPropsType = {
      label: 'SubHeader-label',
      primaryActionLabel: '',
      primaryActionIcon: '',
      primaryAction: null
    };

    const element = render(<SubHeader {...props} />);

    expect(element).toMatchSnapshot();
  });

  test('If SubHeader rendered correctly with primary action', () => {
    const props: SubHeaderPropsType = {
      label: 'SubHeader-label',
      primaryActionLabel: 'SubHeader-action-label',
      primaryActionIcon: 'SubHeader-action-icon.png',
      primaryAction: () => {}
    };

    const element = render(<SubHeader {...props} />);

    expect(element).toMatchSnapshot();
  });

  test('If SubHeader rendered correctly with primary and secondary actions', () => {
    const props: SubHeaderPropsType = {
      label: 'SubHeader-label',
      primaryActionLabel: 'SubHeader-action-label',
      primaryActionIcon: 'SubHeader-action-icon.png',
      primaryAction: () => {},
      secondaryActionLabel: 'SubHeader-action-label',
      secondaryActionIcon: 'SubHeader-action-icon',
      secondaryAction: () => {}
    };

    const element = render(<SubHeader {...props} />);

    expect(element).toMatchSnapshot();
  });
});
