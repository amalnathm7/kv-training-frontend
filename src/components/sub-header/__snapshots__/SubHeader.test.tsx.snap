// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubHeader Props Test If SubHeader rendered correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sub-header"
      >
        <label
          class="sub-header-label"
        >
          SubHeader-label
        </label>
        <div
          class="sub-header-actions"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sub-header"
    >
      <label
        class="sub-header-label"
      >
        SubHeader-label
      </label>
      <div
        class="sub-header-actions"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SubHeader Props Test If SubHeader rendered correctly with primary action 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sub-header"
      >
        <label
          class="sub-header-label"
        >
          SubHeader-label
        </label>
        <div
          class="sub-header-actions"
        >
          <div
            class="action-button"
          >
            <div
              class="sub-header-action-icon-container"
            >
              <img
                class="sub-header-action-icon"
                src="/assets/icons/SubHeader-action-icon.png"
              />
            </div>
            <label
              class="sub-header-action-label"
            >
              SubHeader-action-label
            </label>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sub-header"
    >
      <label
        class="sub-header-label"
      >
        SubHeader-label
      </label>
      <div
        class="sub-header-actions"
      >
        <div
          class="action-button"
        >
          <div
            class="sub-header-action-icon-container"
          >
            <img
              class="sub-header-action-icon"
              src="/assets/icons/SubHeader-action-icon.png"
            />
          </div>
          <label
            class="sub-header-action-label"
          >
            SubHeader-action-label
          </label>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SubHeader Props Test If SubHeader rendered correctly with primary and secondary actions 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sub-header"
      >
        <label
          class="sub-header-label"
        >
          SubHeader-label
        </label>
        <div
          class="sub-header-actions"
        >
          <div
            class="action-button"
          >
            <div
              class="sub-header-action-icon-container"
            >
              <img
                class="sub-header-action-icon"
                src="/assets/icons/SubHeader-action-icon.png"
              />
            </div>
            <label
              class="sub-header-action-label"
            >
              SubHeader-action-label
            </label>
          </div>
          <div
            class="action-button"
          >
            <div
              class="sub-header-action-icon-container"
            >
              <img
                class="sub-header-action-icon"
                src="/assets/icons/SubHeader-action-icon"
              />
            </div>
            <label
              class="sub-header-action-label"
            >
              SubHeader-action-label
            </label>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sub-header"
    >
      <label
        class="sub-header-label"
      >
        SubHeader-label
      </label>
      <div
        class="sub-header-actions"
      >
        <div
          class="action-button"
        >
          <div
            class="sub-header-action-icon-container"
          >
            <img
              class="sub-header-action-icon"
              src="/assets/icons/SubHeader-action-icon.png"
            />
          </div>
          <label
            class="sub-header-action-label"
          >
            SubHeader-action-label
          </label>
        </div>
        <div
          class="action-button"
        >
          <div
            class="sub-header-action-icon-container"
          >
            <img
              class="sub-header-action-icon"
              src="/assets/icons/SubHeader-action-icon"
            />
          </div>
          <label
            class="sub-header-action-label"
          >
            SubHeader-action-label
          </label>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SubHeader Props Test If SubHeader rendered correctly with primary and secondary filter options 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sub-header"
      >
        <label
          class="sub-header-label"
        >
          SubHeader-label
        </label>
        <div
          class="sub-header-actions"
        >
          <div
            class="action-button"
          >
            <div
              class="sub-header-action-icon-container"
            >
              <img
                class="sub-header-action-icon"
                src="/assets/icons/SubHeader-action-icon.png"
              />
            </div>
            <label
              class="sub-header-action-label"
            >
              SubHeader-action-label
            </label>
          </div>
          <div
            class="action-button"
          >
            <div
              class="sub-header-action-icon-container"
            >
              <img
                class="sub-header-action-icon"
                src="/assets/icons/SubHeader-action-icon"
              />
            </div>
            <label
              class="sub-header-action-label"
            >
              SubHeader-action-label
            </label>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sub-header"
    >
      <label
        class="sub-header-label"
      >
        SubHeader-label
      </label>
      <div
        class="sub-header-actions"
      >
        <div
          class="action-button"
        >
          <div
            class="sub-header-action-icon-container"
          >
            <img
              class="sub-header-action-icon"
              src="/assets/icons/SubHeader-action-icon.png"
            />
          </div>
          <label
            class="sub-header-action-label"
          >
            SubHeader-action-label
          </label>
        </div>
        <div
          class="action-button"
        >
          <div
            class="sub-header-action-icon-container"
          >
            <img
              class="sub-header-action-icon"
              src="/assets/icons/SubHeader-action-icon"
            />
          </div>
          <label
            class="sub-header-action-label"
          >
            SubHeader-action-label
          </label>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SubHeader Props Test If SubHeader rendered correctly with primary filter options 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sub-header"
      >
        <label
          class="sub-header-label"
        >
          SubHeader-label
        </label>
        <div
          class="sub-header-actions"
        >
          <div
            class="action-button"
          >
            <div
              class="sub-header-action-icon-container"
            >
              <img
                class="sub-header-action-icon"
                src="/assets/icons/SubHeader-action-icon.png"
              />
            </div>
            <label
              class="sub-header-action-label"
            >
              SubHeader-action-label
            </label>
          </div>
          <div
            class="action-button"
          >
            <div
              class="sub-header-action-icon-container"
            >
              <img
                class="sub-header-action-icon"
                src="/assets/icons/SubHeader-action-icon"
              />
            </div>
            <label
              class="sub-header-action-label"
            >
              SubHeader-action-label
            </label>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sub-header"
    >
      <label
        class="sub-header-label"
      >
        SubHeader-label
      </label>
      <div
        class="sub-header-actions"
      >
        <div
          class="action-button"
        >
          <div
            class="sub-header-action-icon-container"
          >
            <img
              class="sub-header-action-icon"
              src="/assets/icons/SubHeader-action-icon.png"
            />
          </div>
          <label
            class="sub-header-action-label"
          >
            SubHeader-action-label
          </label>
        </div>
        <div
          class="action-button"
        >
          <div
            class="sub-header-action-icon-container"
          >
            <img
              class="sub-header-action-icon"
              src="/assets/icons/SubHeader-action-icon"
            />
          </div>
          <label
            class="sub-header-action-label"
          >
            SubHeader-action-label
          </label>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SubHeader Props Test If SubHeader rendered correctly with route options 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sub-header"
      >
        <select
          class="sub-header-label"
        >
          <option
            value="route 1"
          >
            route 1
          </option>
          <option
            value="route 2"
          >
            route 2
          </option>
        </select>
        <div
          class="sub-header-actions"
        >
          <div
            class="action-button"
          >
            <div
              class="sub-header-action-icon-container"
            >
              <img
                class="sub-header-action-icon"
                src="/assets/icons/SubHeader-action-icon.png"
              />
            </div>
            <label
              class="sub-header-action-label"
            >
              SubHeader-action-label
            </label>
          </div>
          <div
            class="action-button"
          >
            <div
              class="sub-header-action-icon-container"
            >
              <img
                class="sub-header-action-icon"
                src="/assets/icons/SubHeader-action-icon"
              />
            </div>
            <label
              class="sub-header-action-label"
            >
              SubHeader-action-label
            </label>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sub-header"
    >
      <select
        class="sub-header-label"
      >
        <option
          value="route 1"
        >
          route 1
        </option>
        <option
          value="route 2"
        >
          route 2
        </option>
      </select>
      <div
        class="sub-header-actions"
      >
        <div
          class="action-button"
        >
          <div
            class="sub-header-action-icon-container"
          >
            <img
              class="sub-header-action-icon"
              src="/assets/icons/SubHeader-action-icon.png"
            />
          </div>
          <label
            class="sub-header-action-label"
          >
            SubHeader-action-label
          </label>
        </div>
        <div
          class="action-button"
        >
          <div
            class="sub-header-action-icon-container"
          >
            <img
              class="sub-header-action-icon"
              src="/assets/icons/SubHeader-action-icon"
            />
          </div>
          <label
            class="sub-header-action-label"
          >
            SubHeader-action-label
          </label>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SubHeader Props Test If SubHeader rendered correctly with search field 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sub-header"
      >
        <label
          class="sub-header-label"
        >
          SubHeader-label
        </label>
        <div
          class="sub-header-actions"
        >
          <select
            class="filter-dropdown"
            style="margin-right: 20px;"
          >
            <option
              hidden=""
            />
          </select>
          <select
            class="filter-dropdown"
          >
            <option
              hidden=""
            />
            <option
              value="filter 1"
            >
              filter 1
            </option>
            <option
              value="filter 2"
            >
              filter 2
            </option>
          </select>
          <div
            class="action-button"
          >
            <div
              class="sub-header-action-icon-container"
            >
              <img
                class="sub-header-action-icon"
                src="/assets/icons/search.png"
              />
            </div>
            <input
              class="sub-header-action-label"
              placeholder="Subheader-placeholder"
              style="font-size: 14px; cursor: text;"
              value="Subheader-value"
            />
          </div>
          <div
            class="action-button"
          >
            <div
              class="sub-header-action-icon-container"
            >
              <img
                class="sub-header-action-icon"
                src="/assets/icons/SubHeader-action-icon"
              />
            </div>
            <label
              class="sub-header-action-label"
            >
              SubHeader-action-label
            </label>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sub-header"
    >
      <label
        class="sub-header-label"
      >
        SubHeader-label
      </label>
      <div
        class="sub-header-actions"
      >
        <select
          class="filter-dropdown"
          style="margin-right: 20px;"
        >
          <option
            hidden=""
          />
        </select>
        <select
          class="filter-dropdown"
        >
          <option
            hidden=""
          />
          <option
            value="filter 1"
          >
            filter 1
          </option>
          <option
            value="filter 2"
          >
            filter 2
          </option>
        </select>
        <div
          class="action-button"
        >
          <div
            class="sub-header-action-icon-container"
          >
            <img
              class="sub-header-action-icon"
              src="/assets/icons/search.png"
            />
          </div>
          <input
            class="sub-header-action-label"
            placeholder="Subheader-placeholder"
            style="font-size: 14px; cursor: text;"
            value="Subheader-value"
          />
        </div>
        <div
          class="action-button"
        >
          <div
            class="sub-header-action-icon-container"
          >
            <img
              class="sub-header-action-icon"
              src="/assets/icons/SubHeader-action-icon"
            />
          </div>
          <label
            class="sub-header-action-label"
          >
            SubHeader-action-label
          </label>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
